src/components/editor/codeMirror.ts

```diff
@@ -173,12 +172,45 @@ export function createCodeMirrorEditor(element: HTMLElement, cell: Cell, opts: {
 
     const readOnlyExtension = tagExtension('readOnly', EditorView.editable.of(!cell.metadata.properties.locked));
 
+    const cellSwitchExtension = keymap.of([
+        { key: "ArrowUp", run: (target) => { 
+                if(target.state.selection.ranges.length === 1 && target.state.selection.ranges[0].empty) {
+                    let firstLine = target.state.doc.line(1);
+                    let cursorPosition = target.state.selection.ranges[0].head;
+                    if(firstLine.from <= cursorPosition && cursorPosition <= firstLine.to) {
+                        _runtime.controls.emit({id: cell.id, type: "RUN_CELL", focusNext: "before"});
+                        return true
+                    } else {
+                        return false;
+                    }
+                } else {
+                    return false; 
+                }
```

Comment:
Nit: You could replace with to `return false` without any `elses`. A matter of style but I think it's slightly easier to read.